// Proto file representing the EDB TermInstTerm class

syntax = "proto3";

package ansys.api.edb.v1;

import "edb_messages.proto";
import "refs.proto";

service TerminalInstanceTerminalService {
  rpc Create (TermInstTermCreationMessage) returns (EDBObjMessage) {}
  rpc GetTerminalInstance (EDBObjMessage) returns (EDBObjMessage) {}
  rpc SetTerminalInstance (TermInstTermSetInstanceMessage) returns (google.protobuf.Empty) {}
}

message TermInstTermCreationMessage {
  EDBObjMessage layout = 1;
  NetRefMessage net = 2;
  string name = 3;
  EDBObjMessage term_inst = 4;
  bool is_ref = 5;
}

message TermInstTermSetInstanceMessage {
  EDBObjMessage term = 1;
  EDBObjMessage term_inst = 2;
}
